lm20/pipelines.py

```diff
@@ -168,22 +168,37 @@ def file_path(self, request, response=None, info=None, *, item=None):
 
         media_guid = hashlib.sha1(to_bytes(request.url)).hexdigest()
 
-        content_disposition = headers.get("Content-Disposition").decode()
-        _, params = cgi.parse_header(content_disposition)
-        filename = params["filename"]
+        content_disposition = headers.get("Content-Disposition")
+
+        if type(content_disposition) is bytes:
+            content_disposition = content_disposition.decode()
+
+        if content_disposition:
+            _, params = cgi.parse_header(content_disposition)
+            filename = params["filename"]
+
+            media_ext = os.path.splitext(filename)[1]
+        else:
+            filename = ""
+            media_ext = ""
 
-        media_ext = os.path.splitext(filename)[1]
         # Handles empty and wild extensions by trying to guess the
         # mime type then extension or default to empty string otherwise
         if media_ext not in mimetypes.types_map:
             media_ext = ""
             media_type = mimetypes.guess_type(filename)[0]
             if media_type:
                 media_ext = mimetypes.guess_extension(media_type)
-            else:
-                media_ext = mimetypes.guess_extension(
-                    headers.get("Content-Type").decode("utf").split(";")[0]
-                )
+            elif headers.get("Content-Type"):
+                content_type = headers["Content-Type"]
+                if type(content_type) is bytes:
+                    media_ext = mimetypes.guess_extension(
+                        content_type.decode("utf").split(";")[0]
```

Comment:
if it's utf encoded, you do not need to specify that and can just do "content_type.decode()"..